webapp/src/Controller/API/JudgehostController.php

```diff
@@ -828,32 +828,56 @@ public function internalErrorAction(Request $request): ?int
     /**
      * Give back the unjudged runs from the judging with the given judging ID
      * @param int       $judgingId
-     * @param Judgehost $judgehost
+     * @param Judgehost|null $judgehost
      */
-    protected function giveBackJudging(int $judgingId, Judgehost $judgehost)
+    protected function giveBackJudging(int $judgingId, ?Judgehost $judgehost)
     {
         /** @var Judging $judging */
         $judging = $this->em->getRepository(Judging::class)->find($judgingId);
         if ($judging) {
             $this->em->transactional(function () use ($judging, $judgehost) {
                 /** @var JudgingRun $run */
                 foreach ($judging->getRuns() as $run) {
+                    if ($judgehost === null) {
```

Comment:
I don’t follow how this is null for an internal error? Don’t we set it on line 822?

Reply:
Only if we disable the judgehost, e.g. because disk was full.

Reply:
Fixed now.